Editor/StateMachines/ShapeState_DrawHeightShape.cs

```diff
@@ -68,13 +68,20 @@ public override ShapeState DoState(Event evt)
                     case EventType.MouseMove:
                     case EventType.MouseDrag:
                         Ray ray = HandleUtility.GUIPointToWorldRay(evt.mousePosition);
-                        Vector3 heightPoint = Math.GetNearestPointRayRay(tool.m_BB_OppositeCorner, tool.m_Plane.normal,
-                            ray.origin, ray.direction);
-
-                        var deltaPoint = heightPoint - tool.m_BB_OppositeCorner;
-                        deltaPoint = Quaternion.Inverse(tool.m_PlaneRotation) * deltaPoint;
-                        deltaPoint = tool.GetPoint(deltaPoint, evt.control);
-                        tool.m_BB_HeightCorner = tool.m_PlaneRotation * deltaPoint + tool.m_BB_OppositeCorner;
+                        Vector3 heightPoint = Math.GetNearestPointRayRay(tool.m_BB_OppositeCorner, tool.m_Plane.normal, ray.origin, ray.direction);
+                        var dot = Vector3.Dot(tool.m_Plane.normal, ray.direction);
+                        var isHeightPointNaN = float.IsNaN(heightPoint.x) || float.IsNaN(heightPoint.y) || float.IsNaN(heightPoint.z);
+                        if (isHeightPointNaN)
+                        {
+                            tool.m_BB_HeightCorner = tool.m_PlaneRotation * Vector3.zero + tool.m_BB_OppositeCorner;
+                        }
```

Comment:
After reviewing, I would even simplify here to remove a non-necessary multiplication
```suggestion
tool.m_BB_HeightCorner = tool.m_BB_OppositeCorner;
```
I'm even wondering if that still necessary.... I think before going through that we might already have `tool.m_BB_HeightCorner == tool.m_BB_OppositeCorner`.
In that case, we just have to check if (!isHeightPointNaN) and do what's in the else then!

Reply:
Oh cool! Yeah it does work by just checking if (!isHeightPointNaN) and doing what's in the else.

Reply:
Resolved in [b73f4bf](https://github.com/Unity-Technologies/com.unity.probuilder/pull/436/commits/b73f4bf38a164b87902806d0d331ef4ac28d903f)